Automattic/portkey-android
# [DO NOT MERGE] Introduce OnMaxMinFontSizeReached listener (scaling refactor part 5)

This PR adds a listener so the emoji view resizes itself only within the minimum and maximum width/height allowed as per the minimum and maximum fontSize set on the `AutoResizeTextView` containing the emoji.

Before this PR, a view's size could be augmented and take as much as the whole screen (to the extent your fingers would allow to in a pinch-to-zoom gesture). With this PR, the `TextViewSizeAwareTouchListener`  (in charge of handling the touch gestures) will be notified when the `AutoResizeTextView` view used for emoji/text reaches its maximum/minimum allowed `fontSize`, so the `TextViewSizeAwareTouchListener` can check the view's `width`/`height` and stop growing the view (same for shrinking, with the inferior limit).

This detection mechanism will be used next in order to pivot between view resize / view scaling methods in a follow up PR.

Notes:
- This PR merges #206 as I needed to have the maximum fontSize const in place
- Built incrementally on top of #210 
- introduces the AutoResizeTextView class source in the project as we needed to modify it 
4a5f271
- adds a "maximum fontSize reached" listener, as well as a "minimum fontSize reached" listener. 7c5a523

To test:

0. enable developer options -> see layout bounds to see the view's "borders" on your screen
1. take a picture and add an emoji
2. pinch to zoom, observe once it reaches a maximum allowed fontSize (currently 112sp), the view stops growing.
3. if possible do the same when trying to shrink the view (feel free to use some other minimum value in `dimens.xml`  changing these two values to some other, greater ones so it's easier to test, for example 96dp):
```   
    <dimen name="autosize_tv_minimum_width">96dp</dimen>
    <dimen name="autosize_tv_minimum_height">96dp</dimen>
```
